### ISSUE TITLE
The version of mmdet3d

### ISSUE THREAD

**Issue by Jingwei-Jiang** [linxuewu/Sparse4D#7]
Hello, I found that the version of mmdet3d is 0.17 < 1.0, while the latest version is >1.0 with some changes. Is the code suitable for the >1.0 version? (In my observation based on the bbox-coder code, I think yes.) If not, where should I revise, such as dataloader or boxcoder?

**Comment by linxuewu** [comment]
Our code does not rely much on mmdet3d, mainly the box3d structure, but the structure of mmdet3d has not been modified for a long time, so the latest version of mmdet3d is also supported. 
You can try the inference test first. If the metrics are consistent with the published ones, it means that there is no problem.

**Comment by Jingwei-Jiang** [comment]
Thank you. By the way, where is the file "nuscenes_infos_trainval_with_inds.pkl" used? I can't find another usage except for README.md.

**Comment by linxuewu** [comment]
This is for multi-object tracking task, but we haven't released the tracking code yet.

**Comment by BaoWentz** [comment]
> This is for multi-object tracking task, but we haven't released the tracking code yet.

Thanks for your awesome job! Do you have any release plan about tracking task?

**Comment by linxuewu** [comment]
Currently there are no plans.

**Comment by LHY-HongyangLi** [comment]
@linxuewu , Hello, what is the purpose of switching L and W at: https://github.com/linxuewu/Sparse4D/blob/main/projects/mmdet3d_plugin/datasets/nuscenes_3d_det_track_dataset.py#L948

**Comment by linxuewu** [comment]
Because the order defined by mmdet3d is different from that of nuscenes.
https://github.com/open-mmlab/mmdetection3d/blob/master/tools/data_converter/nuscenes_converter.py#L256

**Comment by LHY-HongyangLi** [comment]
But in mmdet3d==0.17.2, there is no such switching 
https://github.com/open-mmlab/mmdetection3d/blob/v0.17.2/tools/data_converter/nuscenes_converter.py#L253

So this means that, you preprocess the nuscenes dataset through the latest mmdet3d but train/evaluate the method in mmdet3d==0.17.2 ?

**Comment by linxuewu** [comment]
It seems so. I didn't pay attention to the mmdet3d version used for data processing before...

**Comment by linxuewu** [comment]
However, training and evaluation do not rely too much on mmdet3d, so version misalignment will not cause other bugs.

**Comment by LHY-HongyangLi** [comment]
It seems that the ground truth of rotation is also different in different mmdet3d versions. So the mAOE seems to be affected too.

**Comment by linxuewu** [comment]
This should be the cause of your mAOE problem, just use the new version of mmdet3d for data preprocess.

**Comment by linxuewu** [comment]
Or directly use the pkl file I provided. https://github.com/linxuewu/Sparse4D/releases/download/v0.0/nuscenes_infos_trainval_with_inds.pkl

**Comment by LHY-HongyangLi** [comment]
OK, but how to use this pkl file? Simply replace "nuscenes_infos_val.pkl" with "nuscenes_infos_trainval_with_inds.pkl" in the config file as shown in the following ?

```
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=4,
    train=dict(
        **data_basic_config,
        ann_file=anno_root + 'nuscenes_infos_train.pkl',
        pipeline=train_pipeline,
        test_mode=False,
    ),
    val=dict(
        **data_basic_config,
        ann_file=anno_root + 'nuscenes_infos_val.pkl',
        pipeline=test_pipeline,
        test_mode=True,
    ),
    test=dict(
        **data_basic_config,
        ann_file=anno_root + 'nuscenes_infos_val.pkl',
        pipeline=test_pipeline,
        test_mode=True,
    ),
)
```

**Comment by linxuewu** [comment]
This is the union of the training set and the validation set, prepared for test-model training. You can split it with nuscenes.utils.split.

**Comment by linxuewu** [comment]
https://github.com/linxuewu/Sparse4D/releases/download/v0.0/nuscenes_infos_train.pkl
If you only want to use train-set, use this pkl.

**Comment by LHY-HongyangLi** [comment]
Is it exists in this repository?  

BTW, I will submit a PR to fix the misalignment of the version of mmdetection3d in preprocessing data and training/evaluation.

**Comment by LHY-HongyangLi** [comment]
I have submitted the PR. The problem mainly lies in your modification of "output_to_nusc_box" to fit the misalignment when you are conducting experiments with mmdet3d==0.17.2 but with the dataset preprocessed by mmdet3d>1.0.0.

https://github.com/linxuewu/Sparse4D/pull/10#issue-1640921473
